### ISSUE TITLE
Bug - Menu - Menu items crop content

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly#6565]
**Describe the problem**
This is an issue with 

- [x] Patternfly 5  

[?] Patternfly 6  

Please give a clear and concise description of the problem. Which components are affected?

Menu, specifically the menuitem.

**How do you reproduce the problem?**

Have any content in a menuitem that isn't text, such as a combobutton or button.

**Expected behavior**

Content of a menu item should not be cut off.

**Is this issue blocking you?**

Yes, we have it as a release blocker.

https://github.com/cockpit-project/cockpit/pull/20367/commits/be07061faaa49140df9053104ce54330be7d8e7a
(We override by setting `.pf-v5-c-menu__item-text` to `overflow: visible`)

**Screenshots**

We have a system/light/dark switcher in Cockpit that's in a menu.

![image](https://github.com/patternfly/patternfly/assets/10246/44dad1bf-2ed9-403e-a8e4-75713fc46a4c)

**What is your environment?**
 - OS: Fedora Linux 40
 - Browser: Firefox, Chrome
 - Version: Latest (Firefox 125.0.1)

**What is your product and what release date are you targeting?**

Cockpit. Today was supposed to be the release. :wink:

**Comment by garrett** [comment]
Note: It _might_ be that someone set `overflow: hidden` to contain floats. This is a very old method to do so with some side effects (clipping content), but there are much better ways to do that, like using `display: flow-root` for block elements with some kind of containment (https://css-tricks.com/display-flow-root/)... or display flex, grid, etc. (but these other ones come with lots of other implications).

**Comment by mattnolting** [comment]
## Core issue
The issue is with `toggle-group` (and other components) whose reduced parent size require extra consideration. It was noted in [6190](https://github.com/patternfly/patternfly/pull/6190#issuecomment-1879756168) and [6202](https://github.com/patternfly/patternfly/issues/6202).

Ample width: 

![Screenshot 2024-05-20 at 3 09 40 PM](https://github.com/patternfly/patternfly/assets/5385435/6075bbe8-895b-4d32-82b1-3b98b842df8b)

Reduced width: 

![Screenshot 2024-05-20 at 3 09 32 PM](https://github.com/patternfly/patternfly/assets/5385435/fbf5f7df-e535-4e66-83a6-72002d187f58)

## Overflow hidden

`overflow: hidden` for a standard menu is set at [198](https://github.com/patternfly/patternfly/blob/v6/src/patternfly/components/Menu/menu.scss#L198) on `pf-v6-c-menu` and at [548](https://github.com/patternfly/patternfly/blob/v6/src/patternfly/components/Menu/menu.scss#L548) on `pf-v6-c-menu__item-text` as contents could otherwise spill out of the menu itself and text is supposed to truncate instead of wrap.

https://github.com/patternfly/patternfly/assets/5385435/6d0552cb-43a9-4fe1-bb69-638ef368f52b

## Behavior

Handling this case is the original intent of [overflow menu](https://pf6.patternfly.org/components/overflow-menu#introduction) (which was created long ago) and [referenced here](https://github.com/patternfly/patternfly-design/issues/1279#issuecomment-2070329669). 

![image](https://github.com/patternfly/patternfly/assets/5385435/bfff9503-f46c-4e3f-a3fe-c0abfb01f5a7)

## Potential solution

In an ideal world we would revisit `overflow-menu` and add the necessary functionality to handle this situation by relying on available size and without the use of breakpoints. This is one instance where [resize observer](https://github.com/patternfly/patternfly/issues/6660) would be particularly useful.

**Comment by garrett** [comment]
However, the problem is with the menu cropping the content, not with the option group. And your demo doesn't show the problem that's indicated in my screenshot... the left side of the select widget begin cropped off. That's the problem, not flowing to the right (which isn't the problem that I'm talking about; we're not seeing that).

I don't see how (or why) adding JavaScript can fix a layout problem with the CSS used in PatternFly. At best, adding more JavaScript (instead of fixing the CSS) would be a workaround that introduces a performance penalty instead of just fixing the problem.

**Comment by mattnolting** [comment]
TLDR: toggle group shouldn't be inside `pf-v6-c-menu__item-text`.

**Comment by garrett** [comment]
OK, what's the advisable way to include the toggle group inside the menu? Is it to drop `pf-v6-c-menu__item-text`?

**Comment by mattnolting** [comment]
> OK, what's the advisable way to include the toggle group inside the menu? Is it to drop `pf-v6-c-menu__item-text`?

That is correct.

**Comment by andrew-ronaldson** [comment]
Catching up on the thread but from a design perspective I think a [flyout menu](https://www.patternfly.org/components/menus/custom-menus#flyout-menu) or a [drilldown](https://www.patternfly.org/components/menus/custom-menus#with-drilldown) would be a cleaner solution inside a menu panel and available without additional modifications.

**Comment by mcoker** [comment]
@garrett did any of the suggestions from Matt/Andrew work for you or do you still need help finding a solution?

**Comment by garrett** [comment]
Still an issue that should be addressed. Not stale.
